text/add-bf16-f64f64-and-f80-type.md

```diff
@@ -0,0 +1,121 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`bf16` is available on all targets. The operators and constants defined for `f32` are also available for `bf16`.
+
+For `f64f64` and `f80`, their availability is limited to the following targets, but this may change over time:
+
+- `f64f64` is supported on `powerpc-*` and `powerpc64(le)-*`, available in `core::arch::{powerpc, powerpc64}`
+- `f80` is supported on `i[356]86-*` and `x86_64-*`, available in `core::arch::{x86, x86_64}`
+
+The operators and constants defined for `f32` and `f64` are available for `f64f64` and `f80` in their respective arch-specific modules.
+
+All proposed types do not have literal representation. Instead, they can be converted to or from IEEE 754 compliant types.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `bf16` type
+
+`bf16` consists of 1 sign bit, 8 bits of exponent, 7 bits of mantissa. Some ARM, AArch64, x86 and x86_64 targets support `bf16` operations natively. For other targets, they will be promoted into `f32` before computation and truncated back into `bf16`.
+
+`bf16` will generate the `bfloat` type in LLVM IR.
+
+## `f64f64` type
+
+`f64f64` is the legacy extended floating point format used on PowerPC targets. It consists of two `f64`s, with the former acting as a normal `f64` and the latter for an extended mantissa.
+
+The following `From` traits are implemented in `core::arch::{powerpc, powerpc64}` for conversion between `f64f64` and other floating point types:
+
+```rust
+impl From<bf16> for f64f64 { /* ... */ }
+impl From<f32> for f64f64 { /* ... */ }
+impl From<f64> for f64f64 { /* ... */ }
+```
+
+`f64f64` will generate `ppc_fp128` type in LLVM IR.
+
+## `f80` type
+
+`f80` represents the extended precision floating point type on x86 targets, with 1 sign bit, 15 bits of exponent and 63 bits of mantissa.
+
+The following `From` traits are implemented in `core::arch::{x86, x86_64}` for conversion between `f80` and other floating point types:
+
+```rust
+impl From<bf16> for f80 { /* ... */ }
+impl From<f32> for f80 { /* ... */ }
+impl From<f64> for f80 { /* ... */ }
+```
+
+`f80` will generate the `x86_fp80` type in LLVM IR.
+
+## `c_longdouble` type in FFI
+
+`core::ffi::c_longdouble` will always represent whatever `long double` does in C. Rust will defer to the compiler backend (LLVM) for what exactly this represents, but it will approximately be:
+
+- `f80` extended precision on `x86` and `x86_64`
+- `f64` double precision with MSVC
+- `f128` quadruple precision on AArch64
+- `f64f64` on PowerPC
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+`bf16` is not an IEEE 754 standard type, so adding it as primitive type may break existing consistency for builtin float types. The truncation after calculations on targets not supporting `bf16` natively also breaks how Rust treats precision loss in other cases.
```

Comment:
I think the idea is that on targets not supporting `bf16`, a technically-incorrect and lazy but fast and sometimes good enough approximation is commonly used: doing the operations as `f32` and then taking the high half of the `f32` result, which has incorrect rounding (that `f32` to `bf16` conversion truncates instead of rounding to nearest, ties to even like all other FP normal operations).